controller/api/Deliveries.php

```diff
@@ -220,4 +211,33 @@ public function publish(ServerRequest $request)
             $this->returnFailure($e);
         }
     }
+
+    private function getDeliveryParameter(?array $requestData): string
+    {
+        if (!array_key_exists(self::REST_DELIVERY_URI, $requestData)) {
+            throw new common_exception_RestApi(__('Missing required parameter: `%s`', self::REST_DELIVERY_URI), 400);
+        }
+        if (empty($requestData[self::REST_DELIVERY_URI])) {
+            throw new common_exception_RestApi(__('Parameter `%s` should not be empty.', self::REST_DELIVERY_URI), 400);
+        }
+
+        return $requestData[self::REST_DELIVERY_URI];
+    }
+
+    private function getRemoteEnvironmentsParameter(?array $requestData): array
```

Comment:
$requestData is array or your code won't work, correct?

Reply:
Correct, fixed